# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../rhythmbox/dbus-fix.patch
# Copyright (C) 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

# this patch fixes compile under >= dbus-0.60

diff -Nur rhythmbox-0.9.2-orig/shell/main.c rhythmbox-0.9.2/shell/main.c
--- rhythmbox-0.9.2-orig/shell/main.c	2005-11-21 10:19:19.000000000 +0000
+++ rhythmbox-0.9.2/shell/main.c	2005-12-20 19:05:23.000000000 +0000
@@ -21,18 +21,21 @@
  */
 
 #include <config.h>
-#include <libintl.h>
-#include <locale.h>
-#include <libgnome/gnome-program.h>
-#include <libgnome/gnome-i18n.h>
-#include <libgnomeui/gnome-ui-init.h>
-#include <gtk/gtk.h>
-#include <gdk/gdkx.h> /* For _get_user_time... */
-#include <glade/glade-init.h>
+
 #include <stdlib.h>
 #include <unistd.h>
 #include <time.h>
 #include <string.h>
+
+#include <libintl.h>
+#include <locale.h>
+#include <glib/gi18n.h>
+#include <gdk/gdkx.h> /* For _get_user_time... */
+#include <gtk/gtk.h>
+#include <glade/glade-init.h>
+#include <libgnome/gnome-program.h>
+#include <libgnomeui/gnome-ui-init.h>
+
 #ifdef HAVE_GSTREAMER
 #include <gst/gst.h>
 #include <gst/gconf/gconf.h>
@@ -254,19 +257,25 @@
 		g_clear_error (&error);
 	} else if (!no_registration) {
 		guint request_name_reply;
+		int flags;
 		DBusGProxy *bus_proxy;
 
 		bus_proxy = dbus_g_proxy_new_for_name (session_bus,
 						       "org.freedesktop.DBus",
 						       "/org/freedesktop/DBus",
 						       "org.freedesktop.DBus");
+#ifndef DBUS_NAME_FLAG_DO_NOT_QUEUE
+		flags = DBUS_NAME_FLAG_PROHIBIT_REPLACEMENT;
+else
+		flags = DBUS_NAME_FLAG_DO_NOT_QUEUE;
+#endif
 		if (!dbus_g_proxy_call (bus_proxy,
 					"RequestName",
 					&error,
 					G_TYPE_STRING,
 					"org.gnome.Rhythmbox",
 					G_TYPE_UINT,
-					DBUS_NAME_FLAG_PROHIBIT_REPLACEMENT,
+					flags,
 					G_TYPE_INVALID,
 					G_TYPE_UINT,
 					&request_name_reply,
